lib/tapioca/sorbet_ext/generic_name_patch.rb

```diff
@@ -13,29 +13,35 @@ def [](*types)
 
         return instances[instance_key] if instances.key?(instance_key)
 
-        singleton_class.prepend(Module.new { define_method(:name) { name } })
+        T.unsafe(self).singleton_class.prepend(Module.new { T.unsafe(self).define_method(:name) { name } })
         instances[instance_key] = self
       end
 
       def type_member(variance = :invariant, fixed: nil, lower: T.untyped, upper: BasicObject)
         super.tap do |type_member|
           type_member.singleton_class.instance_exec do
-            define_method(:fixed) { fixed }
-            define_method(:lower) { lower unless lower == T.untyped }
-            define_method(:upper) { upper unless upper == BasicObject }
+            T.unsafe(self).define_method(:fixed) { fixed }
+            T.unsafe(self).define_method(:lower) { lower unless lower == T.untyped }
+            T.unsafe(self).define_method(:upper) { upper unless upper == BasicObject }
           end
+          T::Private::Abstract::Data.set_default(self, "type_variables", []) << type_member
         end
       end
 
       def type_template(variance = :invariant, fixed: nil, lower: T.untyped, upper: BasicObject)
         super.tap do |type_template|
           type_template.singleton_class.instance_exec do
-            define_method(:fixed) { fixed }
-            define_method(:lower) { lower unless lower == T.untyped }
-            define_method(:upper) { upper unless upper == BasicObject }
+            T.unsafe(self).define_method(:fixed) { fixed }
+            T.unsafe(self).define_method(:lower) { lower unless lower == T.untyped }
+            T.unsafe(self).define_method(:upper) { upper unless upper == BasicObject }
           end
+          T::Private::Abstract::Data.set_default(self, "type_variables", []) << type_template
         end
       end
+
+      def __type_variables
```

Comment:
So it's "internal" but not `private`?

Reply:
This was refactored later so that we don't have to store anything on the constant itself. That's why `GenericTypeRegistry` was created.